apis/v1alpha2/grpcroute_types.go

```diff
@@ -549,6 +549,42 @@ type GRPCRouteFilter struct {
 }
 
 // GRPCBackendRef defines how a GRPCRoute forwards a gRPC request.
+//
+// Note that when a namespace different than the local namespace is specified, a
+// ReferenceGrant object is required in the referent namespace to allow that
+// namespace's owner to accept the reference. See the ReferenceGrant
+// documentation for details.
+//
+// <gateway:experimental:description>
+//
+// When the BackendRef points to a Kubernetes Service, implementations SHOULD honor the
+// appProtocol field if it is set for the target Service Port.
+//
+// Implementations supporting appProtocol MUST recognize the Kubernetes Standard Application Protocols
+// defined in [KEP-3726]. This supports IANA standard service names and extra constants defined
+// in the KEP that have a prefix of "kubernetes.io/". Gateway API MAY define additional
+// constants with the prefix "gateway.networking.k8s.io/"
+//
+// If a Service appProtocol isn't specified an implementation MAY infer the backend
+// protocol through its own means. Implementations MAY infer the protocol from the
+// Route type referring to the backend Service.
+//
+// Implementations MAY support Kubernetes Service BackendRefs that are multiplexing TCP
+// and UDP on the same port. Otherwise implementations MUST set the Route ResolvedRefs
+// condition to False with the "UnsupportedProtocol" Reason with a clear message that
+// multiplexing is not supported.
+//
+// If a Route is not able to send traffic to the backend using the specified protocol then
+// the backend is considered invalid. Implementations MUST set ResolvedRefs condition to
+// False with the "UnsupportedProtocol".
+//
+// Implementations MAY support different combinations of protocol/appProtocol/Route Type.
+// See [GEP-1911] for a table.
```

Comment:
In general I don't like relying on GEPs or KEPs as documentation for concepts like this. GEPs + KEPs tend to get stale over time. I'd rather move this to @youngnick's incoming implementer's guide and out of the spec entirely. Mind creating an issue to track this? In the meantime I'd slightly prefer removing it from the spec, but that doesn't need to block.

Reply:
removed this line

Reply:
Issue: https://github.com/kubernetes-sigs/gateway-api/issues/2453